Add SERVICE_NAME variable for docker compose services

Arnaud-IFEA opened the pull request:
## Changes
This change introduces automatically generated `SERVICE_NAME_<SERVICE>`
environment variables for each service within a Docker Compose deployment.
This allows services to reliably reference each other by name, which is particularly
useful in pull request environments where container names are dynamically suffixed.

- The `newParser` helper now generates and injects these `SERVICE_NAME` variables
   into the environment of all services in the compose file.
- `ApplicationDeploymentJob` is updated to correctly handle and filter these
  new variables during deployment.
- UI components and the `EnvironmentVariableProtection` trait have been updated
to make these generated variables read-only, preventing accidental modification.

example:
```yaml
  db:
    image: mysql:latest
    volumes:
      - db-data:/var/lib/mysql
  frontend:
    build:
      context: .
      dockerfile: ./docker/frontend/Dockerfile
```
regular deployment:

```yaml
  db:
    image: mysql:latest
    volumes:
      - db-data:/var/lib/mysql
    environment:
      - SERVICE_NAME_DB=db    # automatically added
      - SERVICE_NAME_FRONTEND=frontend    # automatically added
  frontend:
    build:
      context: .
      dockerfile: ./docker/frontend/Dockerfile
    environment:
      - SERVICE_NAME_DB=db    # automatically added
      - SERVICE_NAME_FRONTEND=frontend    # automatically added
      - DATABASE_HOST=$SERVICE_NAME_DB    # added by yourself to suit your need
```

PR `#6170` deployment 

```yaml
environment:
  - SERVICE_NAME_DB=db-pr-6170
  - SERVICE_NAME_FRONTEND=frontend-pr-6170
  - DATABASE_HOST=$SERVICE_NAME_DB
```
So you can always connect to other containers internally without public URL even for preview deployment.

## Issues
- fix #4708
- https://discord.com/channels/459365938081431553/1121768420883562526/threads/1346960409210388540

Arnaud-IFEA commented:
I may have messed up  `docs: update changelog` , at first i did not understand what was going on, i made a push force after a rebase on `next`, so...

krzkz94 commented:
Finally! Can't wait for this to be merged, working on something that requires it :)

Arnaud-IFEA commented:
Hello @andrasbacsai ,
any chance that you'll consider this PR ? This feature would allow us to use Coolify (as appears to be the case for other people), really useful when you have all your services in a docker compose.

Thanks !

michnhokn commented:
> Hello @andrasbacsai , any chance that you'll consider this PR ? This feature would allow us to use Coolify (as appears to be the case for other people), really useful when you have all your services in a docker compose.
> 
> Thanks !

Yes please work on this! Same issue here!

andrasbacsai commented:
I will review this PR and add it to the next release.

Cinzya commented:
@Arnaud-IFEA just to give a quick update. Andras and Peak are currently busy (vacation and / or moving) so it can potentially take a little bit longer for this to get merged.

Would you be so kind and prepare a PR in the [Coolify Docs](https://github.com/coollabsio/coolify-docs) for this new feature in the meanwhile?

krzkz94 commented:
@andrasbacsai @Cinzya seems like it didn't make it in 420.7, maybe it was forgotten? :D 

Do you guys have any possible ETA on this? Thanks!

andrasbacsai commented:
> @andrasbacsai @Cinzya seems like it didn't make it in 420.7, maybe it was forgotten? :D
> 
> Do you guys have any possible ETA on this? Thanks!

Sorry, yes. Can you please resolve the conflicts? We added a few things here and there. 😅

Arnaud-IFEA committed:
feat(deployment): add SERVICE_NAME variables for service discovery

This change introduces automatically generated `SERVICE_NAME_<SERVICE>`
environment variables for each service within a Docker Compose deployment.
This allows services to reliably reference each other by name, which is particularly
useful in pull request environments where container names are dynamically suffixed.

- The application parser now generates and injects these `SERVICE_NAME` variables
   into the environment of all services in the compose file.
- `ApplicationDeploymentJob` is updated to correctly handle and filter these
  new variables during deployment.
- UI components and the `EnvironmentVariableProtection` trait have been updated
to make these generated variables read-only, preventing accidental modification.

This commit introduces two new helper functions to standardize resource naming
for pull request deployments:

-  `addPreviewDeploymentSuffix()`: Generates a consistent suffix format (-pr-{id})
   for resource names in preview deployments
-  `generateDockerComposeServiceName()`: Creates SERVICE_NAME environment variables
   for Docker Compose services

Arnaud-IFEA commented:
Hello @Cinzya and @andrasbacsai ,
rebase and merge conflicts done

andrasbacsai commented:
Thank you for the PR!
